### PR TITLE
Initial arrow integration for streaming properties

### PR DESCRIPTION
**Pull request by DarthMax**
- Initial support for gds flight server for streamNodeProperty
- Add test for running streamNodeProperties with and without arrow
- Improve initialization of ArrowQueryRunner

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 multimethod >= 1.0, < 2.0
 neo4j >= 4.4.2, < 5.0
 pandas >= 1.0, < 2.0
+pyarrow
```
Comment (adamnsch) [inline requirements/base.txt:4]: We should specify an explicit interval here

File: graphdatascience/tests/integration/conftest.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 AUTH = None
 if os.environ.get("NEO4J_USER") is not None:
     AUTH = (
-        os.environ.get("NEO4J_USER"),
+        os.environ.get("NEO4J_USER", "DUMMY"),
```
Comment (adamnsch) [inline graphdatascience/tests/integration/conftest.py:15]: I think we can revert this change

File: graphdatascience/graph_data_science.py [hunk 3]
```diff
@@ -57,6 +74,21 @@ def __init__(self, endpoint: Union[str, QueryRunner], auth: Any = None, aura_ds:
             raise InvalidServerVersionError(f"{server_version_string} is not a valid GDS library version")
         self._server_version = ServerVersion(*map(int, server_version_match.groups()))
 
+        if arrow and self._server_version >= ServerVersion(2, 1, 0):
+            try:
+                arrow_info: Series = self._query_runner.run_query("CALL gds.debug.arrow()").squeeze()
+                if arrow_info["running"]:
+                    disable_server_verification = driver._pool.pool_config.trust == "TRUST_ALL_CERTIFICATES"
```
Comment (adamnsch) [inline graphdatascience/graph_data_science.py:81]: As @robsdedude warned us on slack we  cannot really rely on these internals and need to find another way of extracting this info

File: graphdatascience/tests/integration/conftest.py (hunk 2 above)
Comment (adamnsch) [reply graphdatascience/tests/integration/conftest.py:15]: nevermind

File: graphdatascience/graph_data_science.py (hunk 3 above)
Comment (adamnsch) [reply graphdatascience/graph_data_science.py:81]: We will always set this to true for now and hopefully there will be a way to poll in the driver API in the future

File: requirements/base.txt (hunk 1 above)
Comment (adamnsch) [reply requirements/base.txt]: done

**Review by knutwalker — APPROVED** [review]
Nice stuff, looking forward to the next one

File: graphdatascience/graph_data_science.py [hunk 4]
```diff
@@ -72,9 +99,7 @@ def driver_config(self) -> Dict[str, Any]:
...
     @classmethod
-    def from_neo4j_driver(cls: Type[GDS], driver: Driver) -> "GraphDataScience":
-        return cls(cls.create_neo4j_query_runner(driver))
-
-    @staticmethod
-    def create_neo4j_query_runner(driver: Driver, auto_close: bool = False) -> Neo4jQueryRunner:
-        return Neo4jQueryRunner(driver, auto_close=auto_close)
+    def from_neo4j_driver(
+        cls: Type[GDS], driver: Driver, auth: Optional[Tuple[str, str]] = None, arrow: bool = True
+    ) -> "GraphDataScience":
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/graph_data_science.py:104]: We cannot reference `GraphDataScience` directly since it's not fully defined at this point, so we have to use a string instead - that's Python for you :man_shrugging:

File: requirements/base.txt [hunk 5]
```diff
@@ -1,3 +1,4 @@
 multimethod >= 1.0, < 2.0
 neo4j >= 4.4.2, < 5.0
 pandas >= 1.0, < 2.0
+pyarrow >= 7.0, < 8.0
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply requirements/base.txt:4]: Thanks for the info :) I think the tests are failing since Python 3.6 is not compat with pyarrow 7, so we definitely need to lower it anyway. pyarrow 3 sounds good to me, what do you think @DarthMax ? We'll need to add pyarrow major version to our tox test matrix though

File: graphdatascience/graph_data_science.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (knutwalker) [reply graphdatascience/graph_data_science.py:104]: Redefining stringly-typed, I see 🙈

File: graphdatascience/query_runner/arrow_query_runner.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            if encrypted
+            else flight.Location.for_grpc_tcp(host, int(port_string))
+        )
+
+        self._flight_client = flight.FlightClient(location, disable_server_verification=disable_server_verification)
+        self._flight_options = flight.FlightCallOptions()
+
+        if auth:
+            username, password = auth
+            (header, token) = self._flight_client.authenticate_basic_token(username, password)
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/query_runner/arrow_query_runner.py]: done

File: requirements/base.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (adamnsch) [reply requirements/base.txt]: Set to >= 3.0 now. Test matrix changes in https://github.com/neo4j/graph-data-science-client/pull/97
